engine/api/migrate/migrate_headcommit.go

```diff
@@ -0,0 +1,79 @@
+package migrate
+
+import (
+	"context"
+
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/engine/api/entity"
+	"github.com/ovh/cds/engine/api/repositoriesmanager"
+	"github.com/ovh/cds/engine/api/workflow_v2"
+	"github.com/ovh/cds/engine/cache"
+	"github.com/ovh/cds/sdk"
+)
+
+func MigrateHeadCommit(ctx context.Context, db *gorp.DbMap, store cache.Store) error {
+	// Migrate entity with commit HEAD
+	entities, err := entity.LoadUnmigratedHeadEntities(ctx, db)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range entities {
+		tx, err := db.Begin()
+		if err != nil {
+			return sdk.WithStack(err)
+		}
+		e.Head = true
+		if err := entity.Update(ctx, tx, &e); err != nil {
+			_ = tx.Rollback()
+			return err
+		}
+		if err := tx.Commit(); err != nil {
+			_ = tx.Rollback()
+			return sdk.WithStack(err)
+		}
+	}
+
+	// Migrate non repositorywebhook hooks
+	hooks, err := workflow_v2.LoadHeadHookToMigrate(ctx, db)
+	if err != nil {
+		return err
+	}
+	repoCache := make(map[string]*sdk.VCSBranch)
+	hooksToUpdate := make([]sdk.V2WorkflowHook, 0)
+	for _, h := range hooks {
+		repoCacheKey := h.VCSName + "/" + h.RepositoryName
+		defaultBranch, has := repoCache[repoCacheKey]
+		if !has {
+			vcsClient, err := repositoriesmanager.AuthorizedClient(ctx, db, store, h.ProjectKey, h.VCSName)
```

Comment:
Maybe you can skip if the vcs client is no more authorized on the repo.

Reply:
fix on getBranch below